consensus/istanbul/validator/default.go

```diff
@@ -358,8 +358,12 @@ func (valSet *defaultSet) Copy() istanbul.ValidatorSet {
 	}
 
 	newValSet := NewSubSet(addresses, valSet.policy, valSet.subSize).(*defaultSet)
-	_, proposer := newValSet.GetByAddress(valSet.GetProposer().Address())
-	newValSet.proposer.Store(proposer)
+	if proposer := newValSet.GetProposer(); proposer != nil {
+		_, proposer := newValSet.GetByAddress(proposer.Address())
+		newValSet.proposer.Store(proposer)
```

Comment:
Already reported some times. Please refer to the [description](https://github.com/klaytn/klaytn/pull/1600#issue-1347472306). Also, you can reproduce the [same panic](https://github.com/klaytn/klaytn/issues/1206).

Reply:
In other words, the proposer on 361 line is not nil, but the next assigned proposer on 362 can be nil, it can cause panic on 363 line. So I commented on checking the second assigned proposer for nil one more time.

Reply:
Oops, I'm sorry. I misinterpreted your comment. A necessary condition to be nil is that a snapshot should set wrong validators. This is impossible because the parent hash of the fetched block differs if the `extraData` differs and does not contribute to updating the snapshot.

Reply:
Thank you for your kind explanation. I will take a closer look at the code referring your explanation.